src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -130,7 +130,12 @@ public static unsafe string GetAsciiStringNonNullCharacters(this Span<byte> span
             return asciiString;
         }
 
-        public static unsafe string GetAsciiOrUTF8StringNonNullCharacters(this Span<byte> span)
+        public static string GetAsciiOrUTF8StringNonNullCharacters(this Span<byte> span)
+        {
+            return GetAsciiOrUTF8StringNonNullCharacters(span, useLatin1: false);
+        }
+
+        public static unsafe string GetAsciiOrUTF8StringNonNullCharacters(this Span<byte> span, bool useLatin1)
```

Comment:
Maybe add a `GetLatin1StringNonNullCharacters` instead of adding a bool to this path? Changing `value.GetAsciiOrUTF8StringNonNullCharacters(_useLatin1)` to `_useLatin1 ? value.GetLatin1StringNonNullCharacters() : value.GetAsciiOrUTF8StringNonNullCharacters()`. Maybe adding an inlined extension method `value.GetStringNonNullCharacters(_useLatin1)` to keep ternary operator out of calling code.

I ask because in `GetLatin1StringNonNullCharacters` method would have a different structure - you wouldn't need to call TryGetAscii because calling GetLatin1String is an equally fast superset of the ascii case. It also wouldn't need any mention of fallback - failing latin1 is an invalid operation.